src/mgmt/cli_mgmt.c

```diff
@@ -468,7 +468,7 @@ int dc_mgmt_net_cfg(const char *name)
 	/* Query the agent for the CaRT network configuration parameters */
 	rc = get_attach_info(name, &npsrs, &psrs, &sy_info);
 	if (rc != 0)
-		return rc;
+		goto cleanup;
```

Comment:
cleanup: tries to free psrs, however it's uninitialised before this call, and if this call fails I assume it's still unset, so the call to put_attach_info() in cleanup will be accessing uninitialized memory, leading to a segfault.

Reply:
Although it is possible for get_attach_info to return non-zero, and yet set psrs, in that case it's probably best to set it to NULL when it's declared, and change put_attach_info() to check if it's zero.

Reply:
OK, now finally I connected these messages, thanks.
The fix is for:
CID 303425 src/mgmt/cli_mgmt.c:468: Variable psrs going out of scope leaks
Which does complain psrs could be allocated in get_attach_info() but never get freed.
Now my change seems missed the opposite case - it's never got allocated but still try to free it.
Agree, initialize it to NULL and check it before freeing it should do the job.

Reply:
Great, glad you sorted this.  Looks like all the Unit Test issues are resolved now.

It's interesting how Coverity and the fault-injection manage to test the same code but in opposite ways.  Fault injection doesn't have a great deal of coverage yet but I'm working on improving it.